http/middleware.go

```diff
@@ -75,37 +76,60 @@ func (s *shttp) GetUser(ctx context.Context) (django.User, error) {
 }
 
 type CodeWriter struct {
+	*sqlx.Tx
 	code int
+	bool
 	http.ResponseWriter
 }
 
 func (c *CodeWriter) WriteHeader(code int) {
-	c.code = code
+	if code == 700 {
+		err := c.Tx.Rollback()
+		if err != nil {
+			amalgam.LOGGER.Crit(
+				"failed_to_rollback_transaction", "err", errors.ErrorStack(err),
+			)
+		}
+	} else {
+		err := c.Tx.Commit()
+		if err != nil {
+			amalgam.LOGGER.Crit(
+				"failed_to_commit_transaction", "err", errors.ErrorStack(err),
+			)
+			c.bool = true
+			err := c.Tx.Rollback()
+			if err != nil {
+				amalgam.LOGGER.Crit(
+					"failed_to_rollback_transaction",
+					"err", errors.ErrorStack(err),
+				)
+			}
+		}
+	}
+	c.code = 200
 	c.ResponseWriter.WriteHeader(code)
 }
 
+func (c *CodeWriter) Write(resp []byte) (int, error) {
+	if c.bool {
```

Comment:
Rename to .hasError. Also do not do anything in .Write() if .hasError, and overwrite .Close().